packages/sysweb3-keyring/src/trezor/index.ts

```diff
@@ -1,2 +1,632 @@
+    this.hdPath = '';
+    this.paths = {};
+    TrezorConnect.on(DEVICE_EVENT, (event: any) => {
+      if (event.payload.features) {
+        this.model = event.payload.features.model;
+      }
+    });
+    this.init();
+  }
+
+  /**
+   * Initialize Trezor script.
+   *
+   * Trezor Connect raises an error that reads "Manifest not set" if manifest is not provided. It can be either set via manifest method or passed as a param in init method.
+   * @returns true, if trezor was initialized successfully and false, if some error happen
+   */
+
+  private async init() {
+    window.TrezorConnect = TrezorConnect;
+    try {
+      await TrezorConnect.init({
+        manifest: TREZOR_CONNECT_MANIFEST,
+        lazyLoad: true,
+        connectSrc: 'https://connect.trezor.io/9/',
+      });
+      return true;
+    } catch (error) {
+      if (
+        error.message.includes('TrezorConnect has been already initialized')
+      ) {
+        return true;
+      }
+      return false;
+    }
+  }
+
+  /**
+   * This return derivated account info based in params provided.
+   *
+   * @param index - index of account for path derivation
+   * @param slip44 - network slip44 number
+   * @param bip - BIP for derivation. Example: 44, 49, 84
+   * @param coin - network symbol. Example: eth, sys, btc
+   * @returns derivated account info or error
+   */
+
+  public async deriveAccount({
+    index,
+    slip44,
+    bip,
+    coin,
+  }: {
+    index: number;
+    slip44: number | string;
+    bip: number;
+    coin: string;
+  }): Promise<
+    | AccountInfo
+    | {
+        error: string;
+        code?: string;
+      }
+  > {
+    const keypath = `m/${bip}'/${slip44}'/0'/0/${index}`;
+
+    return new Promise((resolve, reject) => {
+      TrezorConnect.getAccountInfo({
+        path: keypath,
+        coin: coin,
+      })
+        .then((response) => {
+          if (response.success) {
+            resolve(response.payload);
+          }
+          // @ts-ignore
+          reject(response.payload.error);
+        })
+        .catch((error) => {
+          console.error('TrezorConnectError', error);
+          reject(error);
+        });
+    });
+  }
+
+  /**
+   * This return account info based in params provided.
+   *
+   * @param coin - network symbol. Example: eth, sys, btc
+   * @param slip44 - network slip44 number
+   * @param hdPath - path derivation. Example: m/44'/57'/0'/0/0
+   * @param index - index of account for path derivation
+   * @returns derivated account info or error
+   */
+
+  public async getAccountInfo({
+    coin,
+    slip44,
+    hdPath,
+    index,
+  }: {
+    coin: string;
+    slip44: string;
+    hdPath?: string;
+    index?: string;
+  }): Promise<AccountInfo> {
+    switch (coin) {
+      case 'sys':
+        this.hdPath = `m/84'/57'/0'`;
+        break;
+      case 'btc':
+        this.hdPath = "m/49'/0'/0'";
+        break;
+      default:
+        this.hdPath = `m/44'/${slip44}'/0'/0/${index ? index : 0}`;
```

Comment:
use bip84 as default derivation path too (we yet don't have descriptors logic so Pali is only compatible with bip84) 
m/84'...